### ISSUE TITLE
BadRequest: CORS-Reason

### ISSUE THREAD

**Issue by Angry-Leprechaun** [sendgrid/docs#2482]
I'm having a problem getting my test application off the ground:
Code:
using System;
using System.Threading.Tasks;
using SendGrid;
using SendGrid.Helpers.Mail;

namespace SendGridTest
{
    class Program
    {
        static void Main(string[] args)
        {
            sendEmail().Wait();
        }

        public static async Task sendEmail()
        {
            var apiKey = Environment.GetEnvironmentVariable("SENDGRID_APIKEY", EnvironmentVariableTarget.User);
            var client = new SendGridClient(apiKey);
            var from = new EmailAddress("donotreply@address.com");
            var subject = "Subject";
            var to = new EmailAddress("to@address.com");
            var body = "Email Body";
            var msg = MailHelper.CreateSingleEmail(from, to, subject, body, "");
            var response = await client.SendEmailAsync(msg);

            Console.WriteLine("------Headers------");
            Console.WriteLine(response.Headers);
            Console.WriteLine("------Status-------");
            Console.WriteLine(response.StatusCode);
            Console.WriteLine("------Body---------");
            Console.WriteLine(response.Body);
            Console.ReadLine();
        }
    }
}

Headers:
X-No-CORS-Reason: https://sendgrid.com/docs/Classroom/Basics/API/cors.html <-- No help
Status:
BadRequest
Body:
System.Net.Http.StreamContent

Any help would be appreciated. I'm sure it's something simple.

**Comment by thinkingserious** [comment]
Hello @Angry-Leprechaun,

Thanks for giving SendGrid a try!

This is a bug. The workaround right now is to pass null instead of an empty string. At first, I wanted to require text and html for this helper, because that improves deliverability, but I've found at least a few cases where it is desired to only have text or html.

In v9.0.9 I added the ability to pass in null in that case, but I failed to account for empty strings. This issue will resolve this: https://github.com/sendgrid/sendgrid-csharp/issues/399

Thanks again and please let us know how we can further assist.

With Best Regards,

Elmer

**Comment by Angry-Leprechaun** [comment]
I changed 

var msg = MailHelper.CreateSingleEmail(from, to, subject, body, "");

to

var msg = MailHelper.CreateSingleEmail(from, to, subject, body, null);

and got the same error. Any other ideas?

**Comment by thinkingserious** [comment]
@Angry-Leprechaun,

Can you please verify the version of the SendGrid SDK you are using?

**Comment by Angry-Leprechaun** [comment]
9.0.8.0 is the version that I'm using. I got around the issue by doing the following:

var msg = MailHelper.CreateSingleEmail(from, to, subject, body, body);

**Comment by thinkingserious** [comment]
Hi @Angry-Leprechaun,

Sweet! Thanks for taking the time to reach back out to us :)

**Comment by manuelguzman** [comment]
Thank you.  In my case it was an empty cc that I was passing.

**Comment by anilbobba1** [comment]
I got this error when trying to use the sendGrid V3 API:
"Result": {
    "StatusCode": "BadRequest",

With no error message listed. 
We looked at every possible cause of error including checking for valid API Keys multiple times and HTML body content and template id (we are using transactional templates); after banging the head on the wall a few times, we started removing the different recipients, to, cc, bcc and it suddenly started working. 
The problem is, we had the same email address listed multiple times in the bcc list; once we removed the duplicate, the Send started to work again. 
SendGrid should do better with including the error message.
